front/types/data_source.ts

```diff
@@ -9,4 +9,5 @@ export type DataSourceType = {
   config?: string;
   dustAPIProjectId: string;
   connector: { id: string; provider: ConnectorProvider } | null;
+  connectorType?: ConnectorType | null;
```

Comment:
As discussed earlier this feels wrong since ConnectorType represents a Connector ?

Reply:
see previous comment.